### PR TITLE
feat: grouping of project level roles in autocomplete

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Screenshot 2024-12-30 at 17 01 31

Details:
* project roles and custom roles are grouped
* project roles (Owner,Member) always go first
* custom project roles are sorted alphabetically
* checkbox is aligned with top of the text

### CODE REVIEW

File: frontend/src/component/common/MultipleRoleSelect/MultipleRoleSelect.tsx [hunk 1]
```diff
@@ -41,30 +61,48 @@ export const MultipleRoleSelect = ({
...
             />
             <StyledRoleOption>
                 <span>{option.name}</span>
                 <span>{option.description}</span>
             </StyledRoleOption>
-        </li>
+        </StyledListItem>
     );
 
+    const sortedRoles = sortItems(roles);
```
Comment (kwasniew) [inline frontend/src/component/common/MultipleRoleSelect/MultipleRoleSelect.tsx:77]: project roles before custom roles and alphabetic sorting of custom roles

File: frontend/src/component/common/MultipleRoleSelect/MultipleRoleSelect.tsx [hunk 2]
```diff
@@ -41,30 +61,48 @@ export const MultipleRoleSelect = ({
...
+                            },
+                        },
+                    },
+                }}
                 multiple
                 disableCloseOnSelect
                 openOnFocus
                 size='small'
                 value={value}
+                groupBy={(option) => {
```
Comment (kwasniew) [inline frontend/src/component/common/MultipleRoleSelect/MultipleRoleSelect.tsx:99]: built-in grouping and styling from MUI component

File: frontend/src/component/common/MultipleRoleSelect/MultipleRoleSelect.tsx [hunk 3]
```diff
@@ -41,30 +61,48 @@ export const MultipleRoleSelect = ({
...
-        </li>
+        </StyledListItem>
     );
 
+    const sortedRoles = sortItems(roles);
+
     return (
         <>
             <Autocomplete
+                slotProps={{
```
Comment (kwasniew) [inline frontend/src/component/common/MultipleRoleSelect/MultipleRoleSelect.tsx:82]: override default list item styling to align to the top and remove left spacing

File: frontend/src/component/common/MultipleRoleSelect/MultipleRoleSelect.tsx [hunk 4]
```diff
@@ -13,6 +13,7 @@ import { RoleDescription } from '../RoleDescription/RoleDescription';
 import { ConditionallyRender } from '../ConditionallyRender/ConditionallyRender';
 
 const StyledRoleOption = styled('div')(({ theme }) => ({
+    paddingTop: theme.spacing(0.75),
```
Comment (kwasniew) [inline frontend/src/component/common/MultipleRoleSelect/MultipleRoleSelect.tsx:16]: minor alignment tweak

**Review by sjaanus — APPROVED** [review]

**Pull request merged**